### PR TITLE
Add a configuration validator for cluster membership providers

### PR DESCRIPTION
**Pull request by ReubenBond**
Validates that a clustering provider has been configured on the client and silo.

### CODE REVIEW

File: src/Orleans.Core/Configuration/Validators/ClientClusteringValidator.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace Orleans.Configuration.Validators
+{
+    internal class ClientClusteringValidator : IConfigurationValidator
+    {
+        internal const string ClusteringNotConfigured =
+            "Clustering has not been configured. Configure clustering using one of the clustering packages, such as: "
+            + "\n  * Microsoft.Orleans.Clustering.AzureStorage for Microsoft Azure Storage"
+            + "\n  * Microsoft.Orleans.Clustering.AdoNet for ADO.NET systems such as SQL Server, MySQL, PostgreSQL, and Oracle"
+            + "\n  * Microsoft.Orleans.Clustering.DynamoDB"
+            + "\n  * Microsoft.Orleans.Clustering.ServiceFabric"
```
Comment (sergeybykov) [inline src/Orleans.Core/Configuration/Validators/ClientClusteringValidator.cs]: ZooKeeper and Consul too?

Comment (ReubenBond) [reply src/Orleans.Core/Configuration/Validators/ClientClusteringValidator.cs]: I can add them, I thought the list might get too long

Comment (sergeybykov) [reply src/Orleans.Core/Configuration/Validators/ClientClusteringValidator.cs]: Never mind then.

Comment (ReubenBond) [reply src/Orleans.Core/Configuration/Validators/ClientClusteringValidator.cs]: Well, there's no real reason to be compact here. I'll update it

Comment (ReubenBond) [reply src/Orleans.Core/Configuration/Validators/ClientClusteringValidator.cs]: It's more important that users see something they can use

**Pull request merged**